packages/generator-blueprints/src/utils/read-static-files.ts

```diff
@@ -35,3 +35,52 @@ export async function readPluginStaticFiles(
 
   return fileMap;
 }
+
+export async function importStaticFilesWithReplacements(
+  source: string,
+  basePath: string,
+  placeholders: Record<string, string>,
+  stringReplacements: Record<string, string>
+): Promise<FileMap<string>> {
+  const files = new FileMap<string>(DsgContext.getInstance.logger);
+
+  const staticFiles = await readPluginStaticFiles(source, basePath);
+
+  for (const item of staticFiles.getAll()) {
+    item.code = replacePlaceholders(item.code, placeholders);
+    item.path = replacePlaceholders(item.path, placeholders);
+
+    item.code = replaceText(item.code, stringReplacements);
+    item.path = replaceText(item.path, stringReplacements);
+
+    const file: IFile<string> = {
+      path: item.path,
+      code: item.code,
+    };
+    await files.set(file);
+  }
+
+  return files;
+}
+
+export function replacePlaceholders(
+  template: string,
```

Comment:
I'd change the name of `template` (also on lines 70, 80 & 85) but it's not a showstopper